src/coreclr/src/vm/methodtablebuilder.cpp

```diff
@@ -11602,6 +11602,23 @@ BOOL MethodTableBuilder::ChangesImplementationOfVirtualSlot(SLOT_INDEX idx)
         // methods.
         if (!fChangesImplementation && (ParentImpl.GetSlotIndex() != idx))
             fChangesImplementation = TRUE;
+
+        if (!fChangesImplementation)
+        {
+            // Search the previous slots in the parent vtable for the same implementation. If it exists and it was
+            // overriden, the ClassLoader::PropagateCovariantReturnMethodImplSlots will propagate the change to the current
+            // slot (idx), so the implementation of it will change.
+
```

Comment:
While I believe this is correct, this algorithm is O(n^2) as ChangesImplementationOfVirtualSlot is called for each vtable slot inherited from the parent type. Particular concerns arise from WPF and WinForms scenarios as many of the control classes have >>100 virtual methods.

I believe we either need to 
1. Don't run this algorithm when propagation will not occur. For instance, we have the `HasVTableMethodImpl` check in `PropagateCovariantReturnMethodImplSlots` and checks for PreserveBaseOverridesAttribute before running the secondary loop that makes it O(N^2).
OR 
2. Make the algorithm here not O(N^2)